backend/clickhouse/logs.go

```diff
@@ -29,6 +30,24 @@ type LogRow struct {
 	SecureSessionId    string
 }
 
+func NewLogRow(ts time.Time, projectID int, sessionID, traceID, spanID string) *LogRow {
```

Comment:
Nice!

The only downside I see to this is that it would be pretty easy to mix up the params for `projectID`, `sessionID`, etc.

I wish Go had named parameters (without needing a struct) since that could solve this pretty elegantly.

One quick and dirty way would be for the parameter to accept the same struct:

```go
func NewLogRow(attrs LogRow) *LogRow {
	uuid := uuid.New().String()
	return &LogRow{
		UUID:            uuid,
		Timestamp:       attrs.Timestamp,
		TraceId:         attrs.TraceId,
		SpanId:          attrs.SpanId,
		SeverityText:    "INFO",
		SeverityNumber:  int32(log.InfoLevel),
		ProjectId:       uint32(attrs.ProjectId),
		SecureSessionId: attrs.SecureSessionId,
	}
}
```


Another idea is to make the `logRow` struct private and create a new struct that defines what someone can actually set:

```go
type LogRowAttrs struct {
	Timestamp       time.Time
	ProjectId       uint32
	TraceId         string
	SpanId          string
	SecureSessionId string
}

func NewLogRow(attrs LogRowAttrs) *logRow {
	uuid := uuid.New().String()
	return &logRow{
		UUID:            uuid,
		Timestamp:       attrs.Timestamp,
		TraceId:         attrs.TraceId,
		SpanId:          attrs.SpanId,
		SeverityText:    "INFO",
		SeverityNumber:  int32(log.InfoLevel),
		ProjectId:       uint32(attrs.ProjectId),
		SecureSessionId: attrs.SecureSessionId,
	}
}
```

Reply:
that's fair. i think a base struct that the `LogRow` inherits is a pretty clean way, with that base struct being the arg for `NewLogRow`. just changed that.